### PR TITLE
Fix Metadata::IsSet, operator==, and operator!=

### LINKED ISSUE #84: Metadata::IsSet bug
The Metadata::IsSet member function checks the value of bits beyond the size of the bits_ vector, producing undefined behavior.

This version should work as expected, but it needs to be put in and a test added.

```c++
  bool IsSet(const MetadataFlag bit) const { 
    if (bit.flag_ >= bits_.size()) {
      return false;
    }
    return bits_[bit.flag_];
  }
```

### PR DESCRIPTION
**Pull request by AndrewGaspar**
## PR Summary

Fixes #84 as identified by @jdolence, and then also fixes a related issues in the comparison operator. I added some new tests, too.

Also fixes up cpplint issues that were pre-existing.

## PR Checklist

- [x] Code passes cpplint
- [x] New features are documented. (no new features)
- [x] Adds a test for any bugs fixed. Adds tests for new features.

### CODE REVIEW

File: src/interface/Metadata.hpp [hunk 1]
```diff
@@ -237,21 +239,42 @@ class Metadata {
...
+    return bit.flag_ < bits_.size() && bits_[bit.flag_];
+  }
 
   // Operators
   bool operator==(const Metadata &b) const {
-    return PackedTuple() == b.PackedTuple();
+    auto const &a = *this;
+
+    // Check extra bits are unset
+    auto const max_bits = std::max(a.bits_.size(), b.bits_.size());
```
Comment (AndrewGaspar) [inline src/interface/Metadata.hpp:251]: Wait, this is wrong. D'oh.

File: tst/unit/test_metadata.cpp [hunk 2]
```diff
@@ -30,14 +30,56 @@ TEST_CASE("Built-in flags are registered", "[Metadata]") {
...
+
+#define PARTHENON_INTERNAL_FOR_FLAG(name) \
+    if (Metadata::name != Metadata::FillGhost) REQUIRE(!m.IsSet(Metadata::name));
+    PARTHENON_INTERNAL_FOREACH_BUILTIN_FLAG
+#undef PARTHENON_INTERNAL_FOR_FLAG
+
+        REQUIRE(m.IsSet(Metadata::FillGhost));
+    }
+
+    GIVEN("Two Equivalent Metadata Structs Are Compared") {
```
Comment (jlippuner) [inline tst/unit/test_metadata.cpp:71]: Maybe also compare Metadata with the same flags set in different orders

File: tst/unit/test_metadata.cpp [hunk 3]
```diff
@@ -30,14 +30,56 @@ TEST_CASE("Built-in flags are registered", "[Metadata]") {
...
+        Metadata a({Metadata::Cell}), b({Metadata::Face});
+
+        b.Unset(Metadata::Face);
+        b.Set(Metadata::Cell);
+
```
Comment (jlippuner) [inline tst/unit/test_metadata.cpp:76]: Maybe also set and unset a flat with a larger int behind it and make sure comparison still works as expected

File: tst/unit/test_metadata.cpp [hunk 4]
```diff
@@ -30,14 +30,56 @@ TEST_CASE("Built-in flags are registered", "[Metadata]") {
...
+        REQUIRE(a == b);
+    }
+
+    GIVEN("Two Different Metadata Structs Are Compared") {
```
Comment (jlippuner) [inline tst/unit/test_metadata.cpp:80]: Maybe compare two Metadata with different numbers of flags and/or multiple flags

Comment (Yurlungur) [inline tst/unit/test_metadata.cpp:80]: Does this test actually catch the segfault @jdolence was seeing? i.e., #84 ? Can we compare two metadata structs where we know the flag sizes are different?

And if this test already does stress this code path, can we say that in a comment?

Comment (AndrewGaspar) [reply tst/unit/test_metadata.cpp:80]: This test does not, but it does test the fix I did to `Metadata::operator==`.

File: tst/unit/test_metadata.cpp (hunk 2 above)
Comment (AndrewGaspar) [reply tst/unit/test_metadata.cpp:71]: This test does that, but I can do it for multiple flags.

File: tst/unit/test_metadata.cpp (hunk 3 above)
Comment (AndrewGaspar) [reply tst/unit/test_metadata.cpp:76]: This test does that.

**Pull request merged**